script/Upgrade.s.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.0;
+
+import { ScriptTools } from "dss-test/ScriptTools.sol";
+
+import "forge-std/Script.sol";
+
+import { CCTPForwarder } from "xchain-helpers/forwarders/CCTPForwarder.sol";
+
+import { ControllerInstance }                   from "../deploy/ControllerInstance.sol";
+import { ForeignControllerInit as ForeignInit } from "../deploy/ForeignControllerInit.sol";
+import { MainnetControllerInit as MainnetInit } from "../deploy/MainnetControllerInit.sol";
+
+contract UpgradeMainnetController is Script {
+
+    using stdJson     for string;
+    using ScriptTools for string;
+
+    function run() external {
+        vm.setEnv("FOUNDRY_ROOT_CHAINID",             "1");
+        vm.setEnv("FOUNDRY_EXPORTS_OVERWRITE_LATEST", "true");
+
+        vm.createSelectFork(getChain("mainnet").rpcUrl);
+
+        console.log("Upgrading mainnet controller...");
+
+        string memory fileSlug = string(abi.encodePacked("mainnet-", vm.envString("ENV")));
+
+        address newController = vm.envAddress("NEW_CONTROLLER");
+        address oldController = vm.envAddress("OLD_CONTROLLER");
+
+        vm.startBroadcast();
+
+        string memory inputConfig = ScriptTools.readInput(fileSlug);
+
+        ControllerInstance memory controllerInst = ControllerInstance({
+            almProxy   : inputConfig.readAddress(".almProxy"),
+            controller : newController,
+            rateLimits : inputConfig.readAddress(".rateLimits")
+        });
+
+        MainnetInit.ConfigAddressParams memory configAddresses = MainnetInit.ConfigAddressParams({
+            freezer       : inputConfig.readAddress(".freezer"),
+            relayer       : inputConfig.readAddress(".relayer"),
+            oldController : oldController
+        });
+
+        MainnetInit.CheckAddressParams memory checkAddresses = MainnetInit.CheckAddressParams({
+            admin      : inputConfig.readAddress(".admin"),
+            proxy      : inputConfig.readAddress(".almProxy"),
+            rateLimits : inputConfig.readAddress(".rateLimits"),
+            vault      : inputConfig.readAddress(".allocatorVault"),
+            psm        : inputConfig.readAddress(".psm"),
+            daiUsds    : inputConfig.readAddress(".daiUsds"),
+            cctp       : inputConfig.readAddress(".cctpTokenMessenger")
+        });
+
+        MainnetInit.MintRecipient[] memory mintRecipients = new MainnetInit.MintRecipient[](1);
+
+        string memory baseInputConfig = ScriptTools.readInput(string(abi.encodePacked("base-", vm.envString("ENV"))));
+
+        address baseAlmProxy = baseInputConfig.readAddress(".almProxy");
+
+        mintRecipients[0] = MainnetInit.MintRecipient({
+            domain        : CCTPForwarder.DOMAIN_ID_CIRCLE_BASE,
+            mintRecipient : bytes32(uint256(uint160(baseAlmProxy)))
+        });
+
+        MainnetInit.upgradeController(controllerInst, configAddresses, checkAddresses, mintRecipients);
+
```

Comment:
I would add the rate limits for the things we need to test: Ethena, Aave and Morpho.

I understand this will change, but the planner needs this now and having it in this script is simplest. We can work out some more generic way to add this later. Perhaps another configurable json file in a future PR.

Reply:
Nevermind this is being done in the planner.